readers/timescale/messages.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package timescale
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/jmoiron/sqlx" // required for DB access
+	"github.com/lib/pq"
+	"github.com/mainflux/mainflux/pkg/errors"
+	"github.com/mainflux/mainflux/pkg/transformers/senml"
+	"github.com/mainflux/mainflux/readers"
+)
+
+const errInvalid = "invalid_text_representation"
+
+const (
+	format = "format"
+	// Table for SenML messages
+	defTable = "messages"
+
+	// Error code for Undefined table error.
+	undefinedTableCode = "42P01"
+)
+
+var errReadMessages = errors.New("failed to read messages from timescale database")
+
+var _ readers.MessageRepository = (*timescaleRepository)(nil)
+
+type timescaleRepository struct {
+	db *sqlx.DB
+}
+
+// New returns new TimescaleSQL writer.
+func New(db *sqlx.DB) readers.MessageRepository {
+	return &timescaleRepository{
+		db: db,
+	}
+}
+
+func (tr timescaleRepository) ReadAll(chanID string, rpm readers.PageMetadata) (readers.MessagesPage, error) {
+	order := "time"
+	format := defTable
+
+	if rpm.Format != "" && rpm.Format != defTable {
+		order = "created"
+		format = rpm.Format
+	}
+
+	q := fmt.Sprintf(`SELECT * FROM %s
+    WHERE %s ORDER BY %s DESC
+	LIMIT :limit OFFSET :offset;`, format, fmtCondition(chanID, rpm), order)
```

Comment:
This can probably be defined in one line

Reply:
done